presto-main/src/main/java/com/facebook/presto/operator/HashBuilderOperator.java

```diff
@@ -197,7 +197,12 @@ private int getAndIncrementPartitionIndex(Lifespan lifespan)
         /**
          * No longer needed
          */
-        DISPOSED
+        DISPOSED,
+
+        /**
+         * Close has been called
+         */
+        CLOSED;
```

Comment:
I think what we could do instead of adding a new state is:
- Consider places where `state` is set to `DISPOSED`: `disposeLookupSourceIfRequested`, `disposeUnspilledLookupSourceIfRequested`
- in those places call `close()` instead of setting  `state` to `DISPOSED` (this is a bit something like LJO is already doing: calling close by itself)
- make HBO#close idempotent (by checking `state==DISPOSED`)

Reply:
After some time for thought, i am slightly more inclined towards your original proposal.. 
Or, maybe, just add to the above list one more bullet:
* rename DISPOSED to CLOSED